src/charm.py

```diff
@@ -917,12 +967,26 @@ def _provider_from_relation(self, relation: Relation):
         """Returns the correct IngressProvider based on a relation."""
         relation_type = _get_relation_type(relation)
         if relation_type is _IngressRelationType.per_app:
-            return self.ingress_per_app
+            # first try to tell if remote is speaking v2
+            if self.ingress_per_appv2.is_ready(relation):
```

Comment:
Any chance `is_ready` could be false positive in the future?
Should we use `try ... get_data` instead?